File: cli-go/internal/infra/project/project.go
package project

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/joacohbc/my-devcontainer-installer/cli-go/internal/core"
	"github.com/joacohbc/my-devcontainer-installer/cli-go/internal/domain"
)

type Paths struct {
	ProjectDir     string
	BuildDir       string
	ComposeFile    string
	DockerfilePath string
	EnvPath        string
}

func ResolveWorkspace(cwd string, config *core.DevcontainerConfig) string {
	if config != nil && config.Workspace != "" {
		return config.Workspace
	}
	return domain.SanitizeDockerName(filepath.Base(cwd), "devcontainer")
}

func ProjectPaths(cwd, workspace string) Paths {
	projectDir := filepath.Join(cwd, ".dc_"+workspace)
	buildDir := filepath.Join(projectDir, "build")
	return Paths{
		ProjectDir:     projectDir,
		BuildDir:       buildDir,
		ComposeFile:    filepath.Join(buildDir, "docker-compose.yml"),
		DockerfilePath: filepath.Join(buildDir, "Dockerfile"),
		EnvPath:        filepath.Join(buildDir, ".env"),
	}
}

func ResolveProjectComposeFile(cwd string) (string, error) {
	cfg, err := domain.LoadConfig(cwd)
	if err != nil {
		return "", err
	}
	workspace := ResolveWorkspace(cwd, cfg)
	paths := ProjectPaths(cwd, workspace)
	if _, statErr := os.Stat(paths.ComposeFile); os.IsNotExist(statErr) {
		return "", fmt.Errorf("no compose file found at %s. Run 'devcontainer-cli' to generate one first", paths.ComposeFile)
	}
	return paths.ComposeFile, nil
}

File: cli/src/infra/project.ts
import * as fs from 'fs';
import * as path from 'path';
import { loadConfig } from '@/domain/config.js';
import { sanitizeDockerName } from '@/domain/validators.js';
import type { DevcontainerConfig } from '@/core/types.js';

export function resolveWorkspace(cwd: string, config?: DevcontainerConfig | null): string {
  const cfg = config !== undefined ? config : loadConfig(cwd);
  return cfg?.workspace ?? sanitizeDockerName(path.basename(cwd));
}

export interface ProjectPaths {
  projectDir: string;
  buildDir: string;
  composeFile: string;
  dockerfilePath: string;
  envPath: string;
}

export function projectPaths(cwd: string, workspace: string): ProjectPaths {
  const projectDir = path.join(cwd, `.dc_${workspace}`);
  const buildDir = path.join(projectDir, 'build');
  return {
    projectDir,
    buildDir,
    composeFile: path.join(buildDir, 'docker-compose.yml'),
    dockerfilePath: path.join(buildDir, 'Dockerfile'),
    envPath: path.join(buildDir, '.env'),
  };
}

export function resolveProjectComposeFile(cwd: string): string {
  const workspace = resolveWorkspace(cwd);
  const paths = projectPaths(cwd, workspace);
  if (!fs.existsSync(paths.composeFile)) {
    throw new Error(`No compose file found at ${paths.composeFile}. Run 'devcontainer-cli' to generate one first.`);
  }
  return paths.composeFile;
}
